### PR TITLE
Kubik Server

### PR DESCRIPTION
**Pull request by Trompettesib**

### CODE REVIEW

**Review by huntears — CHANGES_REQUESTED** [review]
This cannot work, the boostrap.sh script downloads to CubicServer instead of the full binary name,.
Also please have the dockerfile build the project and then use the resulting binary, which will allow for not released docker images.

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+WORKDIR /home/cubic-server
+
+COPY CMakeLists.txt .
+COPY cubic-server cubic-server
+COPY generators generators
+COPY blocks.cmake .
+COPY boost.cmake .
+COPY c_flag_overrides.cmake .
+COPY cxx_flag_overrides.cmake .
```
Comment (huntears) [inline docker/Dockerfile:21]: You already have a dockerignore, just do a `COPY . .`

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
+FROM alpine:latest AS builder
```
Comment (huntears) [inline docker/Dockerfile:1]: Please put that file at the root

File: docker/docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+version: "3"
```
Comment (huntears) [inline docker/docker-compose.yaml:1]: Please put that file at the root

File: docker/Dockerfile (hunk 2 above)
Comment (huntears) [reply docker/Dockerfile:1]: Please pin the alpine version to something more stable than latest

File: docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+COPY blocks.cmake .
+COPY boost.cmake .
+COPY c_flag_overrides.cmake .
+COPY cxx_flag_overrides.cmake .
+
+RUN cmake -S . -B build -DCMAKE_BUILD_TYPE=Release -DUSE_CLANG=true -DNO_GUI=true
+
+RUN cmake --build build -j`nproc --ignore=1`
+
+FROM alpine:latest AS runner
```
Comment (huntears) [inline docker/Dockerfile:27]: Please pin the alpine version to something more stable than latest

File: docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+# curl-dev is needed because the library is not linked statically
+RUN apk add --no-cache \
+    libstdc++ \
+    curl-dev
```
Comment (huntears) [inline docker/Dockerfile:32]: curl-dev might not be needed, curl should be enough (Not sure about that though)

**Review by huntears — CHANGES_REQUESTED** [review]
Also doesn't build on my system
image

File: docker/Dockerfile (hunk 5 above)
Comment (Trompettesib) [reply docker/Dockerfile:32]: It is needed
image

**Review by huntears — APPROVED** [review]

**Review by STMiki — APPROVED** [review]

**Pull request merged**